src/java/org/apache/cassandra/concurrent/SEPExecutor.java

```diff
@@ -21,26 +21,34 @@
 import java.util.List;
 import java.util.concurrent.ConcurrentLinkedQueue;
 import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
 import java.util.concurrent.atomic.AtomicLong;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import org.apache.cassandra.metrics.ThreadPoolMetrics;
+import org.apache.cassandra.utils.MBeanWrapper;
 import org.apache.cassandra.utils.concurrent.SimpleCondition;
 import org.apache.cassandra.utils.concurrent.WaitQueue;
 
 import static org.apache.cassandra.concurrent.SEPWorker.Work;
 
-public class SEPExecutor extends AbstractLocalAwareExecutorService
+public class SEPExecutor extends AbstractLocalAwareExecutorService implements SEPExecutorMBean
 {
+    private static final Logger logger = LoggerFactory.getLogger(SEPExecutor.class);
     private final SharedExecutorPool pool;
 
-    public final int maxWorkers;
+    public final AtomicInteger maxWorkers;
+    MaximumPoolSizeListener updatedMaxWorkers;
```

Comment:
Might be nice to also name it like its callback method, i.e. onUpdatedMaxWorkers, since it only has that function.

Reply:
renamed so the update is called as `maximumPoolSizeListener.onUpdateMaximumPoolSize(newMaximumPoolSize);`